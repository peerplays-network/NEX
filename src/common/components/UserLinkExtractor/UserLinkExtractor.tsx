import Link from "next/link";

type Props = {
  infoString: string;
};

export const UserLinkExtractor = ({ infoString }: Props): JSX.Element => {
  const stringParts = infoString.split(",");

  const getUserLink = (userLink: string, key: number) => {
    const trimedUserLink = userLink.replace(/\s/g, "");
    const userName = trimedUserLink.substring(
      trimedUserLink.indexOf("=") + 1,
      trimedUserLink.lastIndexOf("]")
    );
    return (
      <Link key={key} href={`/user/${userName}`}>
        {userName}
      </Link>
    );
  };

  return (
    <>
      {stringParts.map((stringPart, index) => {
        if (stringPart.includes("userlink")) {
          return getUserLink(stringPart, index);
        } else {
          return <span key={index}>{stringPart}</span>;
        }
      })}
    </>
  );
};
