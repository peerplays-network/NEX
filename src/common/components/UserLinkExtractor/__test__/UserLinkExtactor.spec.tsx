import { render } from "@testing-library/react";
import counterpart from "counterpart";
import React from "react";

import { UserLinkExtractor } from "../UserLinkExtractor";

describe("UserLinkExtractor component", () => {
  describe("UserLinkExtractor component in settings page - membership tab", () => {
    it("renders membership info correctly with correct link", () => {
      const dummyComponentProps = {
        infoString: counterpart.translate(
          `pages.settings.membership.pending_fee_description`,
          {
            name: "[userlink = test-account1]",
            maintenanceInterval: "3600",
            nextMaintenanceTime: "Dec 12 2022 18:30:00",
          }
        ),
      };
      const { asFragment } = render(
        <UserLinkExtractor {...dummyComponentProps} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("UserLinkExtractor component in wallet page - Receive tab", () => {
    it("renders receive instruction correctly with correct link", () => {
      const dummyComponentProps = {
        infoString: counterpart.translate(
          `pages.wallet.receive_selected_asset_instruction`,
          {
            assetSymbol: "TEST",
            account: `[userlink = test-account1]`,
          }
        ),
      };
      const { asFragment } = render(
        <UserLinkExtractor {...dummyComponentProps} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
