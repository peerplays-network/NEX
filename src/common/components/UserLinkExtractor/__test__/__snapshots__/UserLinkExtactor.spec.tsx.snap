// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLinkExtractor component UserLinkExtractor component in settings page - membership tab renders membership info correctly with correct link 1`] = `
<DocumentFragment>
  <span>
    Fees paid by 
  </span>
  <a
    href="/user/test-account1"
  >
    test-account1
  </a>
  <span>
     are divided among the network
  </span>
  <span>
     referrers
  </span>
  <span>
     and registrars once every maintenance interval 3600 seconds). The next maintenance time is Dec 12 2022 18:30:00
  </span>
</DocumentFragment>
`;

exports[`UserLinkExtractor component UserLinkExtractor component in wallet page - Receive tab renders receive instruction correctly with correct link 1`] = `
<DocumentFragment>
  <span>
    To receive TEST send funds to 
  </span>
  <a
    href="/user/test-account1"
  >
    test-account1
  </a>
  <span>
     on the Peerplays blockchain
  </span>
</DocumentFragment>
`;
